test/overseer/heartbeat_test.clj

```diff
@@ -0,0 +1,57 @@
+(ns overseer.heartbeat-test
+  (:require [clojure.test :refer :all]
+            [clj-time.core :as clj-time]
+            [framed.std.time :as std.time]
+            [taoensso.timbre :as timbre]
+            [overseer.api :as api]
+            [overseer.core :as core]
+            [overseer.heartbeat :as heartbeat]
+            [overseer.test-utils :as test-utils]))
+
+(defn test-start-heartbeat [store]
+  (timbre/with-log-level :report
+    (let [config {:heartbeat {:monitor-shutdown false}}
+          old-heartbeat
+          (-> (clj-time/now)
+              (clj-time/minus (clj-time/days 30))
+              std.time/datetime->unix)
+          {job-id :job/id :as job} (test-utils/job {:job/heartbeat old-heartbeat})
+          _ (core/transact-graph store (api/simple-graph job))
+          current-job (atom (core/job-info store job-id))
+
+          heartbeat-fut (heartbeat/start-heartbeat config store current-job)]
+      (try
+        (Thread/sleep 500)
```

Comment:
This seems like a little bit of hax.  It seems like there should be way to abstract away time passing behind like, logical async time, and then trigger/simulate time passing (e.g. by `(put! time-ch ...)`).  Totally not a blocker, but I feel like we'd need to do better to have any real system for time/concurrency testing.